Source/ASButtonNode.mm

```diff
@@ -553,6 +539,118 @@ - (UIAccessibilityTraits)defaultAccessibilityTraits
                       : (UIAccessibilityTraitButton | UIAccessibilityTraitNotEnabled);
 }
 
+#pragma mark - Layout
+
+#if !YOGA
+- (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize
+{
+    UIEdgeInsets contentEdgeInsets;
+    ASButtonNodeImageAlignment imageAlignment;
+    ASLayoutSpec *spec;
+    ASStackLayoutSpec *stack = [[ASStackLayoutSpec alloc] init];
+    {
+        ASLockScopeSelf();
+        stack.direction = _laysOutHorizontally ? ASStackLayoutDirectionHorizontal : ASStackLayoutDirectionVertical;
+        stack.spacing = _contentSpacing;
+        stack.horizontalAlignment = _contentHorizontalAlignment;
+        stack.verticalAlignment = _contentVerticalAlignment;
+        
+        contentEdgeInsets = _contentEdgeInsets;
+        imageAlignment = _imageAlignment;
+    }
+    
+    NSMutableArray *children = [[NSMutableArray alloc] initWithCapacity:2];
+    if (_imageNode.image) {
+        [children addObject:_imageNode];
+    }
+    
+    if (_titleNode.attributedText.length > 0) {
+        if (imageAlignment == ASButtonNodeImageAlignmentBeginning) {
+            [children addObject:_titleNode];
+        } else {
+            [children insertObject:_titleNode atIndex:0];
+        }
+    }
+    
+    stack.children = children;
+    
+    spec = stack;
+    
+    if (UIEdgeInsetsEqualToEdgeInsets(UIEdgeInsetsZero, contentEdgeInsets) == NO) {
+        spec = [ASInsetLayoutSpec insetLayoutSpecWithInsets:contentEdgeInsets child:spec];
+    }
+    
+    if (_backgroundImageNode.image) {
+        spec = [ASBackgroundLayoutSpec backgroundLayoutSpecWithChild:spec background:_backgroundImageNode];
+    }
+    
+    return spec;
+}
+#endif
+
+- (void)updateYogaLayoutIfNeeded
+{
+#if YOGA
+  NSMutableArray<ASDisplayNode *> *children = [[NSMutableArray alloc] initWithCapacity:2];
+  {
+    ASLockScopeSelf();
+
+    // Build up yoga children for button node  again
+    unowned ASLayoutElementStyle *style = [self _locked_style];
+    [style yogaNodeCreateIfNeeded];
+
+    // Setup stack layout values
+    style.flexDirection = _laysOutHorizontally ? ASStackLayoutDirectionHorizontal : ASStackLayoutDirectionVertical;
+
+    // Resolve horizontal and vertical alignment
+    ASButtonNodeResolveHorizontalAlignmentForStyle(style, style.flexDirection, _contentHorizontalAlignment, style.justifyContent, style.alignItems);
+    ASButtonNodeResolveVerticalAlignmentForStyle(style, style.flexDirection, _contentVerticalAlignment, style.justifyContent, style.alignItems);
+
+    // Setup new yoga children
+    if (_imageNode.image != nil) {
+      [_imageNode.style yogaNodeCreateIfNeeded];
```

Comment:
What happens if we don't call `yogaNodeCreateIfNeeded` here?

Reply:
If we don't call it explicitly it will be called eventually within `setYogaParent:` the problem though is that we don't have something like a "Yoga node pending state", so for example if we would not call `yogaNodeCreateIfNeeded` for the Yoga children and we set some property like: `firstChildStyle.margin = ASEdgeInsetsMake(UIEdgeInsetsMake(0, 0, _contentSpacing, 0));`,  `propertyDidChange:` will be called and within that callback method we set the Yoga values on the node, unfortunately the Yoga node was not created at the time we set  the property.